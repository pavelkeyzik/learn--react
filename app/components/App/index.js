import React from 'react';

const App = (props) => React.createElement(
  'div',
  null,
  `Application Component. You're in ${props.appName} and you have ${props.appVersion} version.`,
);

export default React.memo(App);
