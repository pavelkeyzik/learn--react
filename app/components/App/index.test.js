import React from 'react';
import App from './index';
import { render } from 'enzyme';
import renderer from 'react-test-renderer';

describe('App module', () => {
  const appVersion = 1.1;
  const appName = 'App Name';

  it('should render snapshot', () => {
    const component = renderer.create(<App appVersion={appVersion} appName={appName} />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have 2 <mark> tags', () => {
    const component = render(<App appVersion={appVersion} appName={appName}/>);

    expect(component.find('mark')).toHaveLength(2);
  });
});
