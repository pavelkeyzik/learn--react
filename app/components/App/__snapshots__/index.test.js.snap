// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App module should render snapshot 1`] = `
<div>
  Application Component. You are in 
  <mark>
    App Name
  </mark>
   and you have 
  <mark>
    1.1
  </mark>
   version.
</div>
`;
